### ISSUE TITLE
Settings reset on update

### ISSUE THREAD

**Issue by psivesely** [gorhill/uBlock#149]
Hi @gorhill,

This hasn't happened every update, but at least the last two version bumps have caused me to lose all my custom settings, which is quite sad because I had spent some decent time unbreaking sites in default-deny. Do you know what is up with this?

Noah

**Comment by gorhill** [comment]
You need to provide more details. I don't even know which browser you are using. See <https://github.com/gorhill/uBlock/blob/master/CONTRIBUTING.md#what-to-include>.

Also, a reminder that to back up all settings is very simple: a mere click on the _Backup to file_ button in the _Settings_ pane.

**Comment by fowlslegs** [comment]
Apologies, I'm
* running uBlock Origin v0.9.5.0
* on Chromium Version 41.0.2272.118 Built on 8.0, running on Debian 8.0 (64-bit, and
* also using uMatrix, Chameleon, RubberGlove, HTTPS Everywhere, Vimium, and Wasavi.

I do have a backup, but it's from a couple weeks ago and I didn't invest in learning how to fully utilize uBlock except in the last month, so most is lost.

**Comment by gorhill** [comment]
uBlock will save its data using Chrome API `chrome.storage.local`, which as far as uBlock is concerned is an opaque storage, I actually don't know into what this translate on disk.

When you say "all my custom settings", what does that include exactly? Dynamic filtering rules? Whtielist directives? Custom static filters? Settings?

**Comment by fowlslegs** [comment]
It refers to everything. Every configurable setting on uBlock has been reset to it's default state.

**Comment by gorhill** [comment]
Do you use incognito mode?

**Comment by WyohKnott** [comment]
When you install the new version, do you use "Load unpacked extension"?
Because if you had an older one from the store installed, you would get two uBlock Origin installed with different IDs. For example on my config I have ID: cjpalhdlnbpafiamejdnhcphjbkeiagm and ID: hdeccefknbcngnnplnaffponljjbiikg
Of course the second one having a different ID can't read the settings of the first one.

**Comment by gorhill** [comment]
@WyohKnott Good point.

The losing-all-settings symptom matches well re-installing uBlock manually, but at a different path. uBlock needs to be re-installed at **exactly** the same path for the settings to be reused.

**Comment by lewisje** [comment]
I forget, do manually-installed extensions still refuse to uninstall unless you delete the folder they're in?

I had this problem back when I was playing around with ABP (to make it redirect to noop content, as gorhill would later try early in uBlock's development), and that led me to forswear installing unpacked extensions (instead I just pack them and test in Canary; I tested in Stable before Google disallowed off-store extension installation in Stable versions on Windows).

**Comment by fowlslegs** [comment]
@gorhill Seldomly.

@WyohKnott I have been using the Web Store version, but it seems to be keeping pretty good pace with the repo. I would install it from the git source if I didn't have to do that for the several different VMs I run Chromium in (I'm actually running Debian in Qubes).

Anyway, that reminded me that there is a VM that I rarely use and might have my rules still. I removed its networking access right before starting it up--so it couldn't update--and was able to get a little more of my ruleset back. I try to keep the VMs as synced as possible, but this is a lesser used one, so it was a little behind in rules, but also still running 0.9.4.4.

What's interesting is that I then updated and restarted that browser, and when it restarted my configuration was still there. This was the 1 of 4 VMs that didn't have the settings wipe problem! So strange! It should be noted that all four of these VMs are based on an identical template root filesystem--and thus are running an identical Chromium binary.
